Support for setting task Date/Time variables

CTI777 opened the pull request:
Related to https://github.com/Activiti/Activiti/issues/2876

CTI777 committed:
[ACTIVITI-3742] Refactoring of TaskVariable Controllers

CTI777 committed:
[ACTIVITI-3742] Refactoring of Task Controllers

CTI777 committed:
[ACTIVITI-3742] Test added for create / update datetime task variable

CTI777 committed:
[ACTIVITI-3742] Refactoring of test for task variables

CTI777 committed:
[ACTIVITI-3742] refactoring of a test for saving task with variables

CTI777 committed:
[ACTIVITI-3742] Test for checking task complete and variables mapping with date variables

CTI777 committed:
[ACTIVITI-3742] refactoring of the extension file

CTI777 committed:
[ACTIVITI-3742] Errors fix

ffazzini commented on activiti-cloud-services-runtime-bundle/activiti-cloud-services-rest-impl/src/main/java/org/activiti/cloud/services/rest/controllers/TaskVariablesPayloadValidator.java:
```diff
@@ -0,0 +1,65 @@
+                if (value instanceof String) {
+                    try {
+                        payloadVar.setValue(dateFormatterProvider.parse((String)value));
+                    } catch (Exception e) {}
```
i see a lot of empty catch blocks in this class, should not we log something or re-throw an handled exception?

CTI777 replied:
I agree that we have to handle exceptions proper... But in these cases we try to convert String to Date, and if it fails, we simply keep variable as a string

ffazzini commented on activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/TaskVariablesIT.java:
```diff
@@ -195,20 +200,94 @@ public void adminShouldSetGetUpdateTaskVariables() {
     }
+    
+    @Test
+    public void should_Create_Update_DateTimeTaskVariables() throws Exception{
```
minor one, can we stick to the naming "should_..._When_...." ?

CTI777 replied:
Have try to use other names (don't know if they are better)

CTI777 committed:
[ACTIVITI-3742] Change name for test functions

erdemedeiros committed:
Refactor TaskVariablesPayloadValidator

- centralize exception handling
- rename class to TaskVariablesPayloadDateHandler as it's handling dates
- make methods return values to make more explicit that they are modifying values

erdemedeiros commented on activiti-cloud-starter-runtime-bundle/src/test/resources/processes/task-date-var-mapping-extensions.json:
```diff
@@ -0,0 +1,91 @@
+        "name": "process-variable-datetime",
+        "type": "date",
+        "required": true,
+        "value": "2019-09-01"
```
Now you can specify a date with time here, right?

CTI777 replied:
Yes we can... After corresponding PR (activiti-core) was merged, it is possible... Should I apply this here? This check was made in other PR...

erdemedeiros replied:
Please, apply it here as well.. No need to to check the actual value, just change the format (for consistency). You can use zeros to keep the same date if you prefer.

CTI777 replied:
Changed

erdemedeiros reviewed (changes requested):
Just a question.

CTI777 committed:
Update extension file: set date/time value

mergify[bot] committed:
Merge branch 'develop' into dev-CTI777-ACTIVITI-3742-Implement-support-for-setting-task-date-variables
